slides/k8s/setup-managed.md

```diff
@@ -0,0 +1,253 @@
+with a cloud provider
+
+- ... And a credit card
+
+---
+
+## EKS (the hard way)
+
+- [Read the doc](https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html)
+
+- Create service roles, VPCs, and a bunch of other oddities
+
+- Try to figure out why it doesn't work
+
+- Start over, following an [official AWS blog post](https://aws.amazon.com/blogs/aws/amazon-eks-now-generally-available/)
+
+- Try to find the missing Cloud Formation template
+
+--
+
+.footnote[(╯°□°)╯︵ ┻━┻]
+
+---
+
+## EKS (the easy way)
+
+- Install `eksctl`
+
+- Set the usual environment variables
+
+  ([AWS_DEFAULT_REGION](https://docs.aws.amazon.com/general/latest/gr/rande.html#eks_region), AWS_ACCESS_KEY, AWS_SECRET_ACCESS_KEY)
+
+- Create the cluster:
+  ```bash
+  eksctl create cluster
+  ```
+
+- Wait 15-20 minutes (yes, it's sloooooooooooooooooow)
+
+- Add cluster add-ons
+
+  (by default, it doesn't come with metrics-server, logging, etc.)
+
+---
+
+## EKS (cleanup)
+
+- Delete the cluster:
+  ```bash
+  eksctl delete cluster <clustername>
+  ```
+
+- If you need to find the name of the cluster:
+  ```bash
+  eksctl get clusters
+  ```
+
+---
+
+## GKE (initial setup)
+
+- Install `gcloud`
+
+- Login:
+  ```bash
+  gcloud auth init
+  ```
+
+- Create a "project":
+  ```bash
+  gcloud projects create my-gke-project
+  gcloud config set project my-gke-project
+  ```
+
+- Pick a [region](https://cloud.google.com/compute/docs/regions-zones/)
+
+  (example: `europe-west1`, `us-west1`, ...)
+
+---
+
+## GKE (create cluster)
+
+- Create the cluster:
+  ```bash
+  gcloud container clusters create my-gke-cluster --region us-west1 --num-nodes=2
+  ```
+
+  (without `--num-nodes` you might exhaust your IP address quota!)
+
+- The first time you try to create a cluster in a given project, you get an error
+
+  - you need to enable the Kubernetes Engine API
+  - the error message gives you a link
+  - follow the link and enable the API (and billing)
+    <br/>(it's just a couple of clicks and it's instantaneous)
+
+- Wait a couple of minutes (yes, it's faaaaaaaaast)
+
+- The cluster comes with many add-ons
+
+---
+
+## GKE (cleanup)
+
+- List clusters (if you forgot its name):
+  ```bash
+  gcloud container clusters list
+  ```
+
+- Delete the cluster:
+  ```bash
+  gcloud container clusters delete my-gke-cluster --region us-west1
+  ```
+
+- Delete the project (optional):
+  ```bash
+  gcloud projects delete my-gke-project
+  ```
+
+---
+
+## AKS (initial setup)
+
+- Install the Azure CLI
+
+- Login:
+  ```bash
+  az login
+  ```
+
+- Select a [region](https://azure.microsoft.com/en-us/global-infrastructure/services/?products=kubernetes-service&regions=all
+)
+
+- Create a "resource group":
+  ```bash
+  az group create --name my-aks-group --location westeurope
+  ```
+
+---
+
+## AKS (create cluster)
+
+- Create the cluster:
+  ```bash
+  az aks create --resource-group my-aks-group --name my-aks-cluster
+  ```
+
+- Wait 15 minutes (sometimes longer)
```

Comment:
One creation took this long:
Sat Apr 20 11:23:16 CDT 2019
Sat Apr 20 11:28:04 CDT 2019

Reply:
Another creation (in the westeurope region again) took this long:

Sat Apr 20 11:43:40 CDT 2019
Sat Apr 20 11:48:27 CDT 2019

I think it's safe to say "wait about 5 minutes (sometimes longer)" instead of 15 minutes.

Reply:
Thanks for checking this! And it did indeed got much faster in the last few weeks. If you get the opportunity, pass along my regards to whomever is responsible for that improvement! <3